src/main/containers/TestPanel.jsx

```diff
@@ -1,30 +1,70 @@
 import React, { useState } from 'react';
-import DataTree from '../components/DataTree.jsx';
 // sample JSON to pass down as props. Will be able to remove as the project evolves.
-import { largeData, smallData } from '../dummyData';
+// import { largeData, smallData } from '../dummyData';
+import DataTree from '../components/DataTree.jsx';
 import DataCanvas from './DataCanvas.jsx';
 import StyledPanel from './StyledPanel.jsx';
 
 // will need to get data from the get request to pass to the formatted view
 
 const TestPanel = (props) => {
-  const { data, updateTreeCount } = props;
-  const [mockups, setData] = useState([props.data]);
+  const { treeCount, updateTreeCount, data, setTests, tests } = props;
+  const [testsListCounter, setTestsListCounter] = useState(0);
   const dataTreeOptions = {
-    onAdd: false,
-    onEdit: false,
-    onDelete: false,
+    onAdd: true,
+    onEdit: true,
+    onDelete: true,
     enableClipboard: false,
   };
 
+  function saveUpdatedTree (newData, arrayPosition) {
+    const testsClone = [...tests];
+    testsClone[arrayPosition].payload = newData;
+    setTests(testsClone);
+  };
+
+  const testsList = [];
+  let i = 0;
+  tests.forEach((test) => {
+    testsList.push(
+
+      <DataTree
+        treeCount={i}
+        key={'TestPanelDataTree' + i}
+        data={data}
+        options={dataTreeOptions}
+        saveUpdatedTree={saveUpdatedTree}
+      />
+
+    );
+    i += 1;
+  });
+
+  function createNewTest () {
```

Comment:
I’m concerned about performance here. Testing it out, it’s taking upwards of 4–5 seconds for a new test to appear on my machine (data from `https://jsonplaceholder.typicode.com/posts`). It wouldn’t be so bad but the app freezes up, too. No changes for now but let’s prioritize optimizing this post-MVP

Reply:
OK.  If you've got ideas on a O(1) way, I'm all ears